File: src/main/java/com/avant/auth/AuthRepository.java
package com.avant.auth;

import org.springframework.data.mongodb.repository.MongoRepository;
import org.springframework.data.mongodb.repository.Query;

import java.util.List;

public interface AuthRepository extends MongoRepository<AuthToken, String> {
	
	@Query ("{'timeout' : {$lt : ?0}}")
	List<AuthToken> findTimeouted (Long now);
	
}

File: src/main/java/com/avant/auth/AuthRepository.kt
package com.avant.auth

import org.springframework.data.mongodb.repository.MongoRepository
import org.springframework.data.mongodb.repository.Query

interface AuthRepository : MongoRepository<AuthToken, String> {
	
	@Query("{'timeout' : {\$lt : ?0}}")
	fun findTimeouted(now: Long?): List<AuthToken>
	
}
